pkg/scheduler/scheduler.go

```diff
@@ -283,8 +283,14 @@ func filterWorkersByResources(workers []*types.Worker, request *types.ContainerR
 		gpuRequestsMap[gpu] = index
 	}
 
+	requiresGPU := len(gpuRequestsMap) > 0
+
 	filteredWorkers := []*types.Worker{}
 	for _, worker := range workers {
+		if requiresGPU && worker.Gpu == "" {
+			continue
```

Comment:
I still think this is unclear. This handles the case where you're skipping workers where there is no GPU if the request requires a GPU, but it's not clear what happens in the reverse case - that you don't require a GPU and the worker has a GPU. I'd make that explicit